### PR TITLE
Profiling support for compiled runtime

### PR DESCRIPTION
**Pull request by Mats-SX**

### CODE REVIEW

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/CodeGenerator.scala [hunk 1]
```diff
@@ -22,27 +22,28 @@ package org.neo4j.cypher.internal.compiler.v2_3.birk
...
 
-import scala.collection.Map
+import scala.collection.{Map, immutable, mutable}
 
 object CodeGenerator {
   def generateClass(instructions: Seq[Instruction]) = {
     val className = Namer.newClassName()
     val source = generateCodeFromInstructions(className, instructions)
-    //    print(indentNicely(source))
+    println(indentNicely(source))
```
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/CodeGenerator.scala]: remove

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/CodeGenerator.scala [hunk 2]
```diff
@@ -166,6 +166,7 @@ object CodeGenerator {
        |catch (Exception e)
        |{
        |//TODO proper error handling
+       |e.printStackTrace();
```
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/CodeGenerator.scala]: remove

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/CodeGenerator.scala (hunk 1 above)
Comment (Mats-SX) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/CodeGenerator.scala]: Indeed. I actually changed this now to return the source as well (in a tuple). That way the caller can print (or inspect) the source and we don't need to keep un/commenting this line.

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/CodeGenerator.scala (hunk 2 above)
Comment (Mats-SX) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/CodeGenerator.scala]: Done.

**Comment by Mats-SX** [comment]
This PR is for diffing and building purposes. It will be closed after build results are in, and superseded by #4549.

**Pull request closed without merging**